### DISCUSSION TITLE
MacOS: executing 'make' for current xorgxrdp generates errors from not being able to find X11 headers

### DISCUSSION THREAD

**Discussion by myusrn** [neutrinolabs/xorgxrdp#263] [Q&A] [answered]
with the current xorgxrdp-0.9.19 download when i execute 'make' i get the following errors arising from what appears to be non-fully path qualified X11/*.h file references that live in /opt/X11/include/X11 path.

```
In file included from rdpDraw.c:37:
In file included from /opt/X11/include/xorg/xf86.h:44:
In file included from /opt/X11/include/xorg/xf86str.h:37:
/opt/X11/include/xorg/misc.h:74:10: fatal error: 'X11/Xosdefs.h' file not found
#include <X11/Xosdefs.h>
```

I can get these 'make' errors go away if i execute `cp -r /opt/X11/include/X11/* /opt/X11/include/xorg/X11/` and i cannot make them go away if i do things such as `export C_INCLUDE_PATH=/opt/X11/include/:$C_INCLUDE_PATH && export CPLUS_INCLUDE_PATH=/opt/X11/include/X11/:$CPLUS_INCLUDE_PATH`.

q1. Is that work around expected and acceptable or is there some aspect of my xcode and x11 libraries setup, that i put in place following [these](https://github.com/neutrinolabs/xrdp/wiki/Building-on-OSX-(not-official)) macos xrdp build instructions, that i'm overlooking such that the `make` command will work without making header file copies?

q2. When i then execute `sudo make install` i'm unable to find any `xorgxrdp*` build output on my system based on doing an `ls -al /usr/local/sbin && find / -name "xorgxrdp*" 2>&1 | grep -vP "(Permission denied)|(Operation not permitted)|(No such file or directory)"` search for it.  Any pointers on where i should expect files from the the successful xorgxrdp `sudo make install` processing to exist?

**Comment by DJStompZone** [comment, marked as answer]
Hey, sorry I'm late!

I had the same problem. You can resolve this issue by providing the X11 includes path directly when invoking `make`.  For example:

```sh
make CFLAGS="-I/opt/X11/include -I/opt/X11/include/X11"
```

Works on my machine (Macbook Air M2 w/ MacOS Sonoma 14.1.1), so I hope this helps anyone having this problem.

Cheers! :)
